### ISSUE TITLE
False positive `may not support offset operations` for mb_str_split

### ISSUE THREAD

**Issue by Simbiat** [kalessil/phpinspectionsea#1941]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP 8.3                       |

Current behaviour
--
```php
$test = mb_str_split('test', 1, 'UTF-8');
$test[1] = 0;
```
Highlights 2nd line
![image](https://github.com/kalessil/phpinspectionsea/assets/6022665/a8cf901a-8981-411f-a5b4-beca299a604f)

I assume this is because it picks up polyfil
![image](https://github.com/kalessil/phpinspectionsea/assets/6022665/ccfbc767-3e20-4ab8-9904-3335653a091c)
But it should not, since I am not in that namespace, technically. Adding `\` before function does not help.

Expected behavior
No highlight, since only array can be returned by PHP

Environment details
---

```
PhpStorm 2024.1.1
Build #PS-241.15989.102, built on April 23, 2024
Licensed to simbiat.ru / Dmitry Kustov
Subscription is active until May 11, 2024.
For non-commercial open source development only.
Runtime version: 17.0.10+1-b1207.14 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
GC: G1 Young Generation, G1 Old Generation
Memory: 5120M
Cores: 16
Registry:
  debugger.new.tool.window.layout=true
  run.processes.with.pty=TRUE
  ide.experimental.ui=true
Non-Bundled Plugins:
  com.jetbrains.space (241.15989.9)
  com.intellij.ml.llm (241.15989.102)
  com.kalessil.phpStorm.phpInspectionsEA (5.0.0.0)
```

**Comment by Simbiat** [comment]
The same is happening with `array_merge`:
```php
#Full string
$fullStr = array_merge($rkcNum, $account_split);
#Multiplication
for ($i = 0; $i < 23; $i++) {
    $multi[$i] = (int)$fullStr[$i] * $VK[$i];
}
```
`$fullStr[$i]` is being highlighted. In my case there are no other definitions that could have been picked up, and `array_merge` returns only `array` type.

Current environment:
```
PhpStorm 2024.3.4
Build #PS-243.25659.45, built on February 27, 2025
Licensed to Dmitrii Kustov
You have a perpetual fallback license for this version.
Subscription is active until March 3, 2026.
Runtime version: 21.0.6+8-b631.39 amd64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.awt.windows.WToolkit
Windows 11.0
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 5120M
Cores: 16
Registry:
  debugger.new.tool.window.layout=true
  run.processes.with.pty=TRUE
  ide.experimental.ui=true
  terminal.new.ui=true
Non-Bundled Plugins:
  com.intellij.ml.llm (243.23654.270.16)
  ru.adelf.idea.dotenv (2024.3)
  de.espend.idea.php.annotation (11.2.0)
  fr.adrienbrault.idea.symfony2plugin (2024.1.276)
  com.kalessil.phpStorm.phpInspectionsUltimate-mp (2024.3)

```

**Comment by vodevel365** [comment]
Similar problem in class methods and functions, but outside of them everything is ok.

![Image](https://github.com/user-attachments/assets/c3439208-ea43-4542-abfb-1da197bb747d)
